xdsl/tools/xdsl_tool.py

```diff
@@ -0,0 +1,218 @@
+from xdsl.dialects.memref import MemRef
+from xdsl.dialects.mpi import MPI
+from xdsl.dialects.pdl import PDL
+from xdsl.dialects.printf import Printf
+from xdsl.dialects.riscv import RISCV
+from xdsl.dialects.riscv_func import RISCV_Func
+from xdsl.dialects.scf import Scf
+from xdsl.dialects.snitch import Snitch
+from xdsl.dialects.snitch_runtime import SnitchRuntime
+from xdsl.dialects.stencil import Stencil
+from xdsl.dialects.test import Test
+from xdsl.dialects.vector import Vector
+from xdsl.frontend.passes.desymref import DesymrefyPass
+from xdsl.frontend.symref import Symref
+from xdsl.ir import Dialect, MLContext
+from xdsl.parser import Parser
+from xdsl.passes import ModulePass
+from xdsl.transforms.dead_code_elimination import DeadCodeElimination
+from xdsl.transforms.experimental.convert_stencil_to_ll_mlir import (
+    ConvertStencilToLLMLIRPass,
+)
+from xdsl.transforms.experimental.dmp.stencil_global_to_local import (
+    GlobalStencilToLocalStencil2DHorizontal,
+    LowerHaloToMPI,
+)
+from xdsl.transforms.experimental.stencil_shape_inference import (
+    StencilShapeInferencePass,
+)
+from xdsl.transforms.experimental.stencil_storage_materialization import (
+    StencilStorageMaterializationPass,
+)
+from xdsl.transforms.lower_mpi import LowerMPIPass
+from xdsl.transforms.lower_riscv_func import LowerRISCVFunc
+from xdsl.transforms.lower_snitch import LowerSnitchPass
+from xdsl.transforms.lower_snitch_runtime import LowerSnitchRuntimePass
+from xdsl.transforms.mlir_opt import MLIROptPass
+from xdsl.transforms.printf_to_llvm import PrintfToLLVM
+from xdsl.transforms.reconcile_unrealized_casts import ReconcileUnrealizedCastsPass
+from xdsl.transforms.riscv_register_allocation import RISCVRegisterAllocation
+from xdsl.utils.exceptions import ParseError
+
+
+def get_all_dialects() -> list[Dialect]:
+    """Return the list of all available dialects."""
+    return [
+        Affine,
+        Arith,
+        Builtin,
+        Cf,
+        CMath,
+        DMP,
+        FIR,
+        Func,
+        GPU,
+        Linalg,
+        IRDL,
+        LLVM,
+        Math,
+        MemRef,
+        MPI,
+        PDL,
+        Printf,
+        RISCV,
+        RISCV_Func,
+        Scf,
+        Snitch,
+        SnitchRuntime,
+        Stencil,
+        Symref,
+        Test,
+        Vector,
+    ]
+
+
+def get_all_passes() -> list[type[ModulePass]]:
+    """Return the list of all available passes."""
+    return [
+        ConvertStencilToLLMLIRPass,
+        DeadCodeElimination,
+        DesymrefyPass,
+        GlobalStencilToLocalStencil2DHorizontal,
+        LowerHaloToMPI,
+        LowerMPIPass,
+        LowerRISCVFunc,
+        LowerSnitchPass,
+        LowerSnitchRuntimePass,
+        MLIROptPass,
+        PrintfToLLVM,
+        RISCVRegisterAllocation,
+        RISCVLowerArith,
+        StencilShapeInferencePass,
+        StencilStorageMaterializationPass,
+        ReconcileUnrealizedCastsPass,
+    ]
+
+
+class xDSLTool:
```

Comment:
Is there a reason we don't include `split-input-file` to the list of arguments here?
Do you think it should stay in xdsl-opt only?